fv3gfs-physics/fv3gfs/physics/testing/translate.py

```diff
@@ -0,0 +1,354 @@
+    #     9---4---8
+    #     |       |
+    #     1   5   3
+    #     |       |
+    #     6---2---7
+
+    def __init__(self, inputs, rank, layout, *, backend: str):
+        self.backend = backend
+        self.indices = {}
+        self.shape_params = {}
+        self.data = {}
+        for s in Grid.shape_params:
+            self.shape_params[s] = inputs[s]
+            del inputs[s]
+        self.rank = rank
+        self.layout = layout
+        for i, j in Grid.index_pairs:
+            for index in [i, j]:
+                self.indices[index] = inputs[index] + self.fpy_model_index_offset
+                del inputs[index]
+
+        self.data = inputs
+
+    def make_composite_var_storage(
+        self,
+        varname,
+        data3d,
+        shape,
+    ):
+        for s in range(9):
+            self.data[varname + str(s + 1)] = utils.make_storage_data(
+                np.squeeze(data3d[:, :, s]),
+                shape,
+                origin=(0, 0, 0),
+                backend=self.backend,
+            )
+
+    def make_grid_storage(self, pygrid):
+        shape = pygrid.domain_shape_full(add=(1, 1, 1))
+        for key in TranslateGrid.composite_grid_vars:
+            if key in self.data:
+                self.make_composite_var_storage(key, self.data[key], shape)
+                del self.data[key]
+        for key in TranslateGrid.ee_vars:
+            if key in self.data:
+                self.data[key] = np.moveaxis(self.data[key], 0, 2)
+                self.data[key] = utils.make_storage_data(
+                    self.data[key],
+                    (shape[0], shape[1], 3),
+                    origin=(0, 0, 0),
+                    backend=self.backend,
+                )
+        for key, axis in TranslateGrid.edge_var_axis.items():
+            if key in self.data:
+                self.data[key] = utils.make_storage_data(
+                    self.data[key],
+                    shape,
+                    start=(0, 0, pygrid.halo),
+                    axis=axis,
+                    read_only=True,
+                    backend=self.backend,
+                )
+        for key, value in self.data.items():
+            if type(value) is np.ndarray:
+                # TODO: when grid initialization model exists, may want to use
+                # it to inform this
+                istart, jstart = pygrid.horizontal_starts_from_shape(value.shape)
+                logger.debug(
+                    "Storage for Grid variable {}, {}, {}, {}".format(
+                        key, istart, jstart, value.shape
+                    )
+                )
+                origin = (istart, jstart, 0)
+                self.data[key] = utils.make_storage_data(
+                    value,
+                    shape,
+                    origin=origin,
+                    start=origin,
+                    read_only=True,
+                    backend=self.backend,
+                )
+
+    def python_grid(self):
```

Comment:
All `fv3core` import in physics test have been removed except for `Grid` here. Should we make a new classmethod in `GridData` that makes the class with serialized data? Or should we retain `Grid`? We also need a location for either of them.

Reply:
Grid's main responsibility is to convert serialized data, which may or may not include some number of halos and may or may not be staggered, into standardized storages that all have 3 halos and a compute origin at (3, 3). For this reason, Grid should continue to be used in the test code, and should probably live with the translate infrastructure.

Reply:
This makes sense, I had thought that we wanted to get rid of `Grid` all together. If the plan is to keep it alive within the translate infrastructure, I'll go ahead and reshuffle some things.